ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/deduplication/HexOffset.scala

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.deduplication
+
+import java.math.BigInteger
+
+import com.daml.lf.data.Ref
+
+object HexOffset {
+
+  /** Computes the first lexicographical string before [[value]], if it exists.
+    * If there's no such a string (i.e., the input is 00000000...) then it returns `None`.
+    * The string uses the base16 lowercase format as per [[Ref.HexString]] definition.
+    *
+    * Examples:
+    *
+    *   - for 000000000... it returns None
+    *   - for 000001000 it returns Some(000000fff)
+    *   - for 00000ab00 it returns Some(00000aaff)
+    *   - for 00007a900 it returns Some(00007a8ff)
+    */
+  def firstBefore(value: Ref.HexString): Option[Ref.HexString] = {
```

Comment:
```suggestion
  def previous(value: Ref.HexString): Option[Ref.HexString] = {
```

Reply:
Or even `prev`, `dec[rement]`, `minusOne`

Reply:
I will include this in another PR just so I don't have to wait for the build for this PR again before opening the next PR I have inline